### DISCUSSION TITLE
debugtalk中引入第三方依赖，导致hrp执行测试用例失败的解决方法

### DISCUSSION THREAD

**Discussion by kinc123** [httprunner/httprunner#1341] [3 - 分享（Share）]
问题描述：hrp run 执行测试用例失败，报错：ERR [Run] init session runner failed error="init plugin failed: connect grpc plugin failed: Unrecognized remote plugin message: \n\nThis usually means that the plugin is either invalid or simply\nneeds to be recompiled to support the latest protocol."

定位：

经开发者定位，hrp 运行的时候，使用的是 ~/.hrp/venv 虚拟环境。在 debugtalk.py 中引入了第三方依赖，用例执行失败，可在当前环境中安装引入的依赖。

解决方法：

使用Scripts目录下python3下载第三方依赖，例如 C:\Users\Admin\.hrp\venv\Scripts\python3.exe -m pip install urllib3

hrp虚拟环境路径：C:\Users\admin\.hrp\venv     admin即当前用户的家目录。

 urllib3：debugtalk中引入的第三方依赖。

PS：感谢卞雨翔老师和李隆老师帮忙定位问题并且提供解决方案，感谢！

**Comment by debugtalk** [comment]
@kinc123 感谢分享。

在 v4.1.3 版本中，新增支持了通过 `--venv` 指定 Python 虚拟环境；因此你还可以自行准备 venv 依赖，然后再执行测试的时候指定该虚拟环境。

**Reply by Biuuuuu-19** [reply]
我通过../.hrp/venv/pyvenv.vfg 改的include-system-site-packages = true
